services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestsHandler.java

```diff
@@ -448,7 +448,8 @@ private CompletableFuture<ReadResponse> handleMultiGetRequestInParallel(
           if (responseKeySizeList != null) {
             responseKeySizeList.set(subChunkCur, key.keyBytes.remaining());
           }
-          int subPartitionId = getSubPartitionId(key.partitionId, topic, partitionerConfig, key.keyBytes.array());
+          int subPartitionId =
+              getSubPartitionId(key.partitionId, topic, partitionerConfig, ByteUtils.extractByteArray(key.keyBytes));
```

Comment:
I was not involved in the original issue investigation so I'd like to confirm, the extra bytes are in the front or in the back? If it is in the front, then we should use 
extractByteArray(ByteBuffer src, byte[] destination, int offset, int length)
instead based on my understanding...

Reply:
It is in the front, but what is the difference between "extractByteArray(ByteBuffer byteBuffer)" and "extractByteArray(ByteBuffer src, byte[] destination, int offset, int length)" ?

Reply:
extractByteArray(ByteBuffer byteBuffer) I quickly scanned through the impl, it is copying the array from offset 0. I guess we should start from byteBuffer.position()?

Reply:
but anyway saw the comment below, we should try to avoid data copy anyway.

Reply:
Actually the `offset` is for `byte[] destination` not for the array inside bytebuffer